src/index.js

```diff
@@ -149,7 +151,9 @@ module.exports = {
       // ensure we can do basic JSON-RPC over this connection
       this.version(function (errorOrResult) {
         if (errorOrResult instanceof Error || errorOrResult.error) return initialConnectCallback(errorOrResult);
-        this.setupBlockSubscription(function () { initialConnectCallback(null); });
+        this.blockStream = new BlockStream(createBlockStreamTransportAdapter(this), this.BLOCK_POLL_INTERVAL);
+        this.blockStream.subscribe(this.onNewBlock.bind(this));
+        initialConnectCallback(null);
```

Comment:
There is a small (I believe) behavior change here where the connect callback will be called *before* the first block for the subscription has been received.  If this is undesirable, we could change `this.version` a few lines up to `this.getBlockByNumber('latest', false, this.onNextBlock)`.  I didn't do this because I don't think anyone actually cares about blocks right now, and users who care about blocks should call `ethrpc.blockStream.subscribe(callback)` rather than watching the blocks on ethrpc.